Automattic/jetpack
# Search: Add 'My Jetpack' and 'Settings' links to Search plugin line on plugins page

This PR adds both a 'settings' and a 'My Jetpack' link to the plugins page for the Search plugin. It only displays 'my Jetpack' if there is an active connection, and only displays 'settings' if Search is active. 

Fixes #25382 

I am interested in specifically the following things:

* code placement: I am new to working in the plugin, and wondering if this is the best location for the code. I noticed that, for example, Boost handles this in their `class-admin.php` file. We don't have as much structure yet, but it doesn't seem worth restructuring into new classes for just the sake of this one link. 

* using `new Connection_Manager()` here, is this the best approach? Could I do it without creating a new instance and I've missed how? I feel like I could be missing a really obvious better solution. 

#### Other information:


#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* On a site with a Jetpack Search subscription and the Search plugin active, go to `/wp-admin/plugins.php` and check that the links alongside Jetpack Search are `My Jetpack` `Settings` and `Deactivate` 

![image](https://user-images.githubusercontent.com/30754158/185536438-17e44de6-7c97-44bc-aa29-73928fabfa09.png)

* Try deactivating the Search plugin and ensure the only link displayed is `Activate`

* Try disconnecting Jetpack, with the Search plugin active, and check the links displayed are only `Settings` and `Deactivate`
